crates/ruff_cli/src/cache.rs

```diff
@@ -114,16 +203,32 @@ pub fn get<P: AsRef<Path>>(
 }
 
 /// Set a value in the cache.
-pub fn set<P: AsRef<Path>>(
```

Comment:
Why remove this generic? (Not objecting, looking to learn.)

Reply:
I can revert the change if you want. 
I experimented with reading the `source` in `get` instead of serializing it to the cache (because it seems a bit unnecessary) but then forgot to undo the change when I decided to not pursue this approach. 

I ran into problems that `P` is not `Copy` and the constraint that `P` for `path` and `project_dir` must be of the same type was also too restricting.

Reply:
Don't feel strongly, up to you!